avro4s-core/src/test/scala/com/sksamuel/avro4s/record/decoder/SealedTraitDecoderTest.scala

```diff
@@ -1,125 +1,144 @@
-//    Decoder[Buy].decode(ImmutableRecord(schema, Vector(ImmutableRecord(appleschema, Vector(java.lang.Double.valueOf(0.3)))))) shouldBe Buy(Apple(0.3))
-//    Decoder[Buy].decode(ImmutableRecord(schema, Vector(ImmutableRecord(orangeschema, Vector(new Utf8("bright orange")))))) shouldBe Buy(Orange("bright orange"))
-//  }
-//
-//  test("use @AvroNamespace and @AvroName with sealed traits of case objects") {
-//    val thingySchema = SchemaBuilder.enumeration("thingy").symbols("whim_wham", "widget")
-//    val schema = SchemaBuilder.record("ThingHolder").fields().name("thing").`type`(thingySchema).noDefault().endRecord()
-//    val record1 = new GenericData.Record(schema)
-//    record1.put("thing", "whim_wham")
-//
-//    val record2 = new GenericData.Record(schema)
-//    record2.put("thing", "widget")
-//
-//    Decoder[ThingHolder].decode(record1) shouldBe ThingHolder(WhimWham)
-//    Decoder[ThingHolder].decode(record2) shouldBe ThingHolder(Widget)
-//  }
-//
-//  test("use @AvroNamespace and @AvroName with sealed traits of case objects in a round trip") {
-//    val thingySchema = SchemaBuilder.enumeration("thingy").symbols("whim_wham", "widget")
-//    val schema = SchemaBuilder.record("ThingHolder").fields().name("thing").`type`(thingySchema).noDefault().endRecord()
-//
-//    val value = ThingHolder(WhimWham)
-//    val encodedRecord: GenericRecord = Encoder[ThingHolder].encode(value).asInstanceOf[GenericRecord]
-//    val decoded = Decoder[ThingHolder].decode(encodedRecord)
-//    decoded shouldBe value
-//  }
-//
-//}
-//
-//sealed trait Wibble
-//case class Wobble(str: String) extends Wibble
-//case class Wabble(dbl: Double) extends Wibble
-//case class Wrapper(wibble: Wibble)
-//
-//sealed trait Tibble
-//case class Tobble(str: String, place: String) extends Tibble
-//case class Tabble(str: Double, age: Int) extends Tibble
-//case class Trapper(tibble: Tibble)
-//
-//sealed trait Nibble
-//case class Nobble(str: String, place: String) extends Nibble
-//case class Nabble(str: String, age: Int) extends Nibble
-//case class Napper(nibble: Nibble)
-//
-//sealed trait Fruit
-//
-//@AvroNamespace("market")
-//final case class Apple(weight: Double) extends Fruit
-//
-//@AvroNamespace("market")
-//final case class Orange(color: String) extends Fruit
-//
-//@AvroNamespace("market")
-//final case class Buy(fruit: Fruit)
-//
-//final case class ThingHolder(thing: Thingy)
-//
-//@AvroName("thingy")
-//sealed trait Thingy
-//@AvroName("whim_wham")
-//case object WhimWham extends Thingy
-//@AvroName("widget")
-//case object Widget extends Thingy
+package com.sksamuel.avro4s.record.decoder
+
+import com.sksamuel.avro4s._
+import org.apache.avro.SchemaBuilder
+import org.apache.avro.generic.{GenericData, GenericRecord}
+import org.apache.avro.util.Utf8
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+
+class SealedTraitDecoderTest extends AnyFunSuite with Matchers {
+
+  test("support sealed trait enums") {
```

Comment:
This passed before but not after ?
Or just commented it out by mistake ?

Reply:
Sorry, my PR description was a bit terse. I was on my way out the door at the time. To summarise properly:

* The actual fix is the removal of the `&& ctx.isEnum` check in decoders/sealedtraits.scala
* This test file was completely commented out because the tests hadn't been updated to the latest API and thus didn't compile
* I uncommented the tests and updated them to make them compile, but they still failed so I commented them out again
* There was no test specifically for the 'enum encoded as sealed trait + case objects' case, so I added this one
	* There are tests like `use @AvroNamespace and @AvroName with sealed traits of case objects`, which also cover this behaviour, but those tests fail because the annotation handling hasn't been implemented yet

Reply:
Ok thanks.